common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -37,6 +61,21 @@ class VerticalBlock(SequenceFields, XModuleFields, StudioEditableBlock, XmlParse
 
     show_in_read_only_mode = True
 
+    def get_completables(self):
+        """
+        Return a set of descendent blocks that this vertical should mark
+        complete upon viewing.
+        """
+        completion_service = self.runtime.service(self, 'completion')
+        if completion_service is None:
+            return set()
+        # pylint: disable=no-member
+        completables = {block.location for block in self.get_display_items() if is_completable_by_viewing(block)}
```

Comment:
Same questions/concerns as earlier about rollout.  Can this be waffle protected?  Maybe we can discuss elsewhere.